app/src/ui/open-pull-request/pull-request-diff.tsx

```diff
@@ -0,0 +1,96 @@
+import * as React from 'react'
+import { IDiff, ImageDiffType } from '../../models/diff'
+import { Repository } from '../../models/repository'
+import { CommittedFileChange } from '../../models/status'
+import { SeamlessDiffSwitcher } from '../diff/seamless-diff-switcher'
+import { Dispatcher } from '../dispatcher'
+import { openFile } from '../lib/open-file'
+
+interface IPullRequestDiffProps {
+  readonly repository: Repository
+  readonly dispatcher: Dispatcher
+
+  /** The file whose diff should be displayed. */
+  readonly selectedFile: CommittedFileChange | null
+
+  /** The diff that should be rendered */
+  readonly diff: IDiff | null
+
+  /** The type of image diff to display. */
+  readonly imageDiffType: ImageDiffType
+
+  /** Whether we should display side by side diffs. */
+  readonly showSideBySideDiff: boolean
+
+  /** Whether we should hide whitespace in diff. */
+  readonly hideWhitespaceInDiff: boolean
+}
+
+/**
+ * A component for viewing the file diff for a pull request.
+ */
+export class PullRequestDiff extends React.Component<
+  IPullRequestDiffProps,
+  {}
+> {
+  /**
+   * Opens a binary file in an the system-assigned application for
+   * said file type.
+   */
+  private onOpenBinaryFile = (fullPath: string) => {
+    openFile(fullPath, this.props.dispatcher)
+  }
+
+  /** Called when the user changes the hide whitespace in diffs setting. */
+  private onHideWhitespaceInDiffChanged = (hideWhitespaceInDiff: boolean) => {
+    const { selectedFile } = this.props
+    return this.props.dispatcher.onHideWhitespaceInHistoryDiffChanged(
+      hideWhitespaceInDiff,
+      this.props.repository,
+      selectedFile as CommittedFileChange
+    )
+  }
+
+  /**
+   * Called when the user is viewing an image diff and requests
+   * to change the diff presentation mode.
+   */
+  private onChangeImageDiffType = (imageDiffType: ImageDiffType) => {
+    this.props.dispatcher.changeImageDiffType(imageDiffType)
+  }
+
+  private renderDiff() {
+    const { diff, selectedFile } = this.props
+
+    if (diff === null || selectedFile === null) {
+      return
+    }
+
+    const {
+      repository,
+      imageDiffType,
+      hideWhitespaceInDiff,
+      showSideBySideDiff,
+    } = this.props
+
+    return (
+      <SeamlessDiffSwitcher
+        repository={repository}
+        imageDiffType={imageDiffType}
+        file={selectedFile}
+        diff={diff}
+        readOnly={true}
+        hideWhitespaceInDiff={hideWhitespaceInDiff}
+        showSideBySideDiff={showSideBySideDiff}
+        onOpenBinaryFile={this.onOpenBinaryFile}
+        onChangeImageDiffType={this.onChangeImageDiffType}
+        onHideWhitespaceInDiffChanged={this.onHideWhitespaceInDiffChanged}
+      />
+    )
+  }
+
+  public render() {
```

Comment:
My plan is that this component will house the file list for the diff as well. I think the empty change set may need to be handled higher up - more of a "This PR doesn't have changes" blank state for the dialog because when we get to having tabs, we won't want to show the body input tab either in this scenario.

Reply:
... changed the component name. :D..